keel-sql/src/main/resources/db/changelog/20201116-foreign-keys.yml

```diff
@@ -0,0 +1,213 @@
+        onDelete: CASCADE
+- changeSet:
+    id: foreign-keys-on-current_constraint
+    author: fletch
+    changes:
+    - addForeignKeyConstraint:
+        baseTableName: current_constraint
+        baseColumnNames: environment_uid
+        constraintName: fk_current_constraint_environment
+        referencedTableName: environment
+        referencedColumnNames: uid
+        referencesUniqueColumn: true
+        onDelete: CASCADE
+    - addForeignKeyConstraint:
+        baseTableName: current_constraint
+        baseColumnNames: constraint_uid
+        constraintName: fk_current_constraint_environment_artifact_constraint
+        referencedTableName: environment_artifact_constraint
+        referencedColumnNames: uid
+        referencesUniqueColumn: true
+        onDelete: CASCADE
+- changeSet:
+    id: foreign-keys-on-delivery_config_artifact
+    author: fletch
+    changes:
+    - sql:
+        sql: delete from delivery_config_artifact where not exists (select 1 from delivery_config where delivery_config_artifact.delivery_config_uid = delivery_config.uid)
+    - sql:
+        sql: delete from delivery_config_artifact where not exists (select 1 from delivery_artifact where delivery_config_artifact.artifact_uid = delivery_artifact.uid)
+    - addForeignKeyConstraint:
+        baseTableName: delivery_config_artifact
+        baseColumnNames: delivery_config_uid
+        constraintName: fk_delivery_config_artifact_delivery_config
+        referencedTableName: delivery_config
+        referencedColumnNames: uid
+        referencesUniqueColumn: true
+        onDelete: CASCADE
+    - addForeignKeyConstraint:
+        baseTableName: delivery_config_artifact
+        baseColumnNames: artifact_uid
+        constraintName: fk_delivery_config_artifact_delivery_artifact
+        referencedTableName: delivery_artifact
+        referencedColumnNames: uid
+        referencesUniqueColumn: true
+        onDelete: CASCADE
+- changeSet:
+    id: foreign-keys-on-delivery_config_last_checked
+    author: fletch
+    changes:
+    - sql:
+        sql: delete from delivery_config_last_checked where not exists (select 1 from delivery_config where delivery_config_last_checked.delivery_config_uid = delivery_config.uid)
+    - addForeignKeyConstraint:
+        baseTableName: delivery_config_last_checked
+        baseColumnNames: delivery_config_uid
+        constraintName: fk_delivery_config_last_checked_delivery_config
+        referencedTableName: delivery_config
+        referencedColumnNames: uid
+        referencesUniqueColumn: true
+        onDelete: CASCADE
+- changeSet:
+    id: foreign-keys-on-environment
+    author: fletch
+    changes:
+    - addForeignKeyConstraint:
+        baseTableName: environment
+        baseColumnNames: delivery_config_uid
+        constraintName: fk_environment_delivery_config
+        referencedTableName: delivery_config
+        referencedColumnNames: uid
+        referencesUniqueColumn: true
+        onDelete: CASCADE
```

Comment:
Just checking: the explicit deletes we have in the code at the moment should continue to work after this, right?

Reply:
Yes, although they're probably unnecessary